sw/device/tests/sim_dv/inject_scramble_seed.c

```diff
@@ -0,0 +1,121 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/arch/device.h"
+#include "sw/device/lib/base/abs_mmio.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_base.h"
+#include "sw/device/lib/dif/dif_otp_ctrl.h"
+#include "sw/device/lib/dif/dif_rstmgr.h"
+#include "sw/device/lib/runtime/hart.h"
+#include "sw/device/lib/runtime/irq.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/otp_ctrl_testutils.h"
+#include "sw/device/lib/testing/rand_testutils.h"
+#include "sw/device/lib/testing/rstmgr_testutils.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+#include "sw/device/lib/testing/test_framework/status.h"
+
+#include "flash_ctrl_regs.h"
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "otp_ctrl_regs.h"
+
+static dif_rstmgr_t rstmgr;
+static dif_otp_ctrl_t otp_ctrl;
+
+#define WORD_SIZE sizeof(uint64_t)
+
+OTTF_DEFINE_TEST_CONFIG();
+
+bool test_main(void) {
+  dif_rstmgr_reset_info_bitfield_t info;
+  CHECK_DIF_OK(dif_rstmgr_init(
+      mmio_region_from_addr(TOP_EARLGREY_RSTMGR_AON_BASE_ADDR), &rstmgr));
+
+  CHECK_DIF_OK(dif_otp_ctrl_init(
+      mmio_region_from_addr(TOP_EARLGREY_OTP_CTRL_CORE_BASE_ADDR), &otp_ctrl));
+
+  info = rstmgr_testutils_reason_get();
```

Comment:
nit: unless I missed something this test only expects `kDifRstmgrResetInfoPor`, right?
```suggestion
  CHECK(rstmgr_testutils_reason_get() == kDifRstmgrResetInfoPor);
  rstmgr_testutils_reason_clear();
```

Reply:
that's right, see my point above. I don't actually need the sw reset anymore because the bootstrap protocol from the host side issues a reset.  I will get rid of this.